cli/src/commands/split.rs

```diff
@@ -196,26 +200,37 @@ The remainder will be in the second commit.
         commit_builder.write(tx.repo_mut())?
     };
 
-    // Mark the commit being split as rewritten to the second commit. As a
-    // result, if @ points to the commit being split, it will point to the
-    // second commit after the command finishes. This also means that any
-    // bookmarks pointing to the commit being split are moved to the second
-    // commit.
+    // Mark the first commit as the rewrite of the target commit. Since the
+    // first commit inherits the change id of the target commit, this ensures
+    // bookmarks continue pointing to the same change id before and after the
+    // split.
     tx.repo_mut()
-        .set_rewritten_commit(commit.id().clone(), second_commit.id().clone());
+        .set_rewritten_commit(commit.id().clone(), first_commit.id().clone());
     let mut num_rebased = 0;
     tx.repo_mut()
         .transform_descendants(vec![commit.id().clone()], |mut rewriter| {
             num_rebased += 1;
             if args.parallel {
-                rewriter
-                    .replace_parent(second_commit.id(), [first_commit.id(), second_commit.id()]);
+                rewriter.replace_parent(first_commit.id(), [first_commit.id(), second_commit.id()]);
+            } else {
+                rewriter.replace_parent(first_commit.id(), [second_commit.id()]);
             }
-            // We don't need to do anything special for the non-parallel case
-            // since we already marked the original commit as rewritten.
             rewriter.rebase()?.write()?;
             Ok(())
         })?;
+    // Move the working copy commit (@) to the second commit for any workspaces
+    // where the target commit is the working copy commit.
+    for workspace_id in tx
+        .base_repo()
```

Comment:
nit: `.base_repo().clone()` also works in place of `.to_owned()` + `.collect_vec()`. Cloning `Arc` should be cheap.